syscalls_darwin.go

```diff
@@ -159,7 +159,16 @@ func (t *tunReadCloser) Write(from []byte) (int, error) {
 	}
 	t.wBuf = t.wBuf[:len(from)+4]
 
-	t.wBuf[3] = 2 // Family: IP (2)
+	// Determine the IP Family for the NULL L2 Header
+	ipVer := from[0] >> 4
```

Comment:
After a quick test it looks like Linux returns `EIO` when a zero-length buffer is written (regardless if the `IFF_NO_PI` flag is set or not).

My suggestion is to match Linux's behavior and return an error for consistency.